src/basic/terminal-util.c

```diff
@@ -1249,12 +1249,38 @@ int vt_default_utf8(void) {
         return parse_boolean(b);
 }
 
+int vt_verify_kbmode(int fd) {
+        int curr_mode;
+
+        /*
+         * Make sure we only adjust consoles in K_XLATE or K_UNICODE mode.
+         * Otherwise we would (likely) interfere with X11's processing of the
+         * key events.
+         *
+         * http://lists.freedesktop.org/archives/systemd-devel/2013-February/008573.html
+         */
+
+        if (ioctl(fd, KDGKBMODE, &curr_mode) < 0)
+                return -errno;
+
+        return IN_SET(curr_mode, K_XLATE, K_UNICODE) ? 0 : -EBUSY;
+}
+
 int vt_reset_keyboard(int fd) {
-        int kb;
+        int kb, r;
 
         /* If we can't read the default, then default to unicode. It's 2017 after all. */
         kb = vt_default_utf8() != 0 ? K_UNICODE : K_XLATE;
 
+        r = vt_verify_kbmode(fd);
+        if (r < 0) {
+                if (r == -EBUSY) {
+                        log_debug_errno(-r, "Keyboard is not in XLATE or UNICODE mode, not resetting: %m");
```

Comment:
no need to negate "r". log_debug_errno() (and its friends) generally do that internally if necessary.

Let's keep the indentation level small here, hence maybe write as:

```c
if (r == -EBUSY) {
        …
} else if (r < 0)
        return r;
```